### PR TITLE
Switch `Logger.print()` to output to stdout

### PR DESCRIPTION
**Pull request by happz**
As discussed in https://github.com/teemtee/tmt/issues/2106, `Logger.print()` would be the way to emit the actual output of tmt. Unlike logging, this output would go to `stdout`, and will ignore `--quiet` option.

### CODE REVIEW

File: tests/core/env/test.sh [hunk 1]
```diff
@@ -57,7 +57,7 @@ rlJournalStart
         rlPhaseEnd
 
         rlPhaseStartTest "Empty environment file ($execute)"
-            rlRun -s "tmt run -r -e @empty.yaml 2>&1"
+            rlRun -s "tmt run -r -e @empty.yaml"
```
Comment (thrix) [inline tests/core/env/test.sh:60]: @happz can we also test that the quiet flag now silences the `warning` as the user wanted for the proof?

File: tmt/utils.py [hunk 2]
```diff
@@ -1151,12 +1151,21 @@ def _indent(
...
-        """ Print a message regardless the quiet mode """
-        self._logger.print(key, value=value, color=color, shift=shift)
+        """
+        Print out an output.
+
+        This method is supposed to be used for emitting a command output. Not
+        to be mistaken with logging - errors, warnings, general command progress,
+        and so on.
+
+        ``print()`` emits even when ``-quiet`` is used, as the option supresses
```
[the remark this answers is not in the document]
Comment (thrix) [reply tmt/utils.py:1164]: Also. 3rd person present: `suppresses` ? typo?

**Review by thrix — CHANGES_REQUESTED** [review]
Found some silly typos and a ask for testing

File: tests/core/env/test.sh (hunk 1 above)
Comment (happz) [reply tests/core/env/test.sh:60]: It does not yet work this way, yet. This PR points `print()` to `stdout`, there will be another patch changing how `--quiet` works. I added a test though, in 5e471f82, it's commented out.

File: tmt/utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (happz) [reply tmt/utils.py]: Fixed in 5e471f82

**Review by thrix — APPROVED** [review]
LGTM

**Review by janhavlin — APPROVED** [review]

**Review by therazix — APPROVED** [review]

**Review by psss — APPROVED** [review]
Looks good, thanks!

**Pull request merged**